model/patch/github.go

```diff
@@ -97,49 +104,57 @@ type githubIntent struct {
 // BSON fields for the patches
 // nolint
 var (
-	documentIDKey  = bsonutil.MustHaveTag(githubIntent{}, "DocumentID")
-	msgIDKey       = bsonutil.MustHaveTag(githubIntent{}, "MsgID")
-	createdAtKey   = bsonutil.MustHaveTag(githubIntent{}, "CreatedAt")
-	repoNameKey    = bsonutil.MustHaveTag(githubIntent{}, "RepoName")
-	prNumberKey    = bsonutil.MustHaveTag(githubIntent{}, "PRNumber")
-	userKey        = bsonutil.MustHaveTag(githubIntent{}, "User")
-	baseHashKey    = bsonutil.MustHaveTag(githubIntent{}, "BaseHash")
-	patchURLKey    = bsonutil.MustHaveTag(githubIntent{}, "PatchURL")
-	processedKey   = bsonutil.MustHaveTag(githubIntent{}, "Processed")
-	processedAtKey = bsonutil.MustHaveTag(githubIntent{}, "ProcessedAt")
-	intentTypeKey  = bsonutil.MustHaveTag(githubIntent{}, "IntentType")
+	documentIDKey   = bsonutil.MustHaveTag(githubIntent{}, "DocumentID")
+	msgIDKey        = bsonutil.MustHaveTag(githubIntent{}, "MsgID")
+	createdAtKey    = bsonutil.MustHaveTag(githubIntent{}, "CreatedAt")
+	baseRepoNameKey = bsonutil.MustHaveTag(githubIntent{}, "BaseRepoName")
+	headRepoNameKey = bsonutil.MustHaveTag(githubIntent{}, "HeadRepoName")
+	prNumberKey     = bsonutil.MustHaveTag(githubIntent{}, "PRNumber")
+	userKey         = bsonutil.MustHaveTag(githubIntent{}, "User")
+	headHashKey     = bsonutil.MustHaveTag(githubIntent{}, "HeadHash")
+	diffURLKey      = bsonutil.MustHaveTag(githubIntent{}, "DiffURL")
+	processedKey    = bsonutil.MustHaveTag(githubIntent{}, "Processed")
+	processedAtKey  = bsonutil.MustHaveTag(githubIntent{}, "ProcessedAt")
+	intentTypeKey   = bsonutil.MustHaveTag(githubIntent{}, "IntentType")
 )
 
 // NewGithubIntent return a new github patch intent.
-func NewGithubIntent(msgDeliveryID, repoName string, prNumber int, user, baseHash, url string) (Intent, error) {
+func NewGithubIntent(msgDeliveryID string, prNumber int, baseRepoName, headRepoName, headHash, user, url string) (Intent, error) {
```

Comment:
I don't think so, because we'd be going from go-github struct to our struct to patch intent, which seems a little silly.

Alternatively, I can have this function accept go-github's pull request struct?